### PR TITLE
DOC: Fix SA01 errors for Index.hasnans, Index.map, Index.nbytes

### PR DESCRIPTION
**Pull request by shriyakalakata**
- [x] xref #58068 
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: pandas/core/indexes/base.py [hunk 1]
```diff
@@ -2423,6 +2423,13 @@ def hasnans(self) -> bool:
         -------
         bool
 
+        See Also
+        --------
+        Series.isna : Return a boolean same-sized object indicating if the values
```
Comment (mroeschke) [inline pandas/core/indexes/base.py:2428]: Could you reference `Index` instead of `Series` in this file?

File: pandas/core/indexes/base.py [hunk 2]
```diff
@@ -2423,6 +2423,13 @@ def hasnans(self) -> bool:
         -------
         bool
 
+        See Also
+        --------
+        Index.isna : Return a boolean same-sized object indicating if the values
+            are NA.
+        Index.dropna : Return a new Series with missing values removed.
```
Comment (mroeschke) [inline pandas/core/indexes/base.py:2430]:
```suggestion
        Index.dropna : Return a new Index with missing values removed.
```

File: pandas/core/indexes/base.py [hunk 3]
```diff
@@ -6067,6 +6074,11 @@ def map(self, mapper, na_action: Literal["ignore"] | None = None):
             If the function returns a tuple with more than one element
             a MultiIndex will be returned.
 
+        See Also
+        --------
+        Index.apply : Invoke function on values of Series.
```
Comment (mroeschke) [inline pandas/core/indexes/base.py:6079]: It doesn't appear these methods exist on `Index`

Comment (shriyakalakata) [reply pandas/core/indexes/base.py:6079]: Oh, sorry about that. Updating the methods!

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @shriyakalakata

**Pull request merged**